src/scripts/storage/clipperData.ts

```diff
@@ -0,0 +1,49 @@
+import {GetResponseAsync, TimeStampedData} from "../http/cachedHttp";
+import {ClipperCachedHttp} from "../http/clipperCachedHttp";
+
+import {Logger} from "../logging/logger";
+
+import {ClipperStorageGateStrategy} from "./clipperStorageGateStrategy";
+import {Storage} from "./storage";
+import {StorageGateStrategy} from "./StorageGateStrategy";
+
+/**
+ * The data-access-like object that handles the fetching and caching of data
+ * from Clipper-specific endpoints (TODO: a factory class that is responsible for
+ * providing getRemoteValue objects?). Also provides a gating mechanism as a
+ * sanity-check for what information gets stored locally.
+ */
+export class ClipperData implements Storage {
+	private storage: Storage;
+	private storageGateStrategy: StorageGateStrategy;
+	private cachedHttp: ClipperCachedHttp;
+
+	constructor(storage: Storage, logger?: Logger) {
+		this.storage = storage;
+		this.storageGateStrategy = new ClipperStorageGateStrategy(storage);
```

Comment:
Mapping keys to strategies is an interesting idea ... I'll have a think about it. As for dependency injection, I decided to hardcode the implementation into the constructor itself as it seems like a 1:1 relationship, and if we constructor-inject it, I was thinking that we'd expose the implementation to the outside world, which is unwanted. I can definitely see the benefits of DI, which makes me wonder if we should use a DI framework some time down the road :P
I'll ponder your comments and make changes as necessary. Thanks!